### ISSUE TITLE
`type[None]` in union type is incorrectly evaluated to `None`

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#2179]
## Environment data

-   Language Server version: 2021.12.3-pre.1
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.12 Anaconda

## Sample code

https://mypy-play.net/?mypy=latest&python=3.9&gist=a98182f0aed5fe7cbb9701adc560470b

```python
def f() -> "type[str] | type[None]":
    ...

r = f()
if r is not None:
    _ = reveal_type(r)
```

For a more practical case, see the inferred types of `typing._type_check` which never returns `None`.

## Expected behavior

Type of `r` is "Type[str] | NoneType" initially and remains the same in the `if` block.

## Actual behaviour

Type of `r` is "Type[str] | None" initially and becomes "Type[str]" in the `if` block.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.1.0, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202210-5-january-2022)
